mne/viz/_brain/_brain.py

```diff
@@ -2496,6 +2541,157 @@ def add_foci(self, coords, coords_as_verts=False, map_surface=None,
+            color = _assign_by_name_or_ch_type(colors, ch_name, ch_type)
+            scale = _assign_by_name_or_ch_type(scales, ch_name, ch_type)
+            alpha = _assign_by_name_or_ch_type(alphas, ch_name, ch_type)
+            if color is None:
+                color = 'y'  # default
+                if ch_type + '_color' in DEFAULTS['coreg']:
+                    color = DEFAULTS['coreg'][ch_type + '_color']
+            if alpha is None:
+                alpha = 0.5
+            color = colorConverter.to_rgba(color, alpha)
+            if scale is None:
+                scale = 5  # default
+                if ch_type + '_scale' in DEFAULTS['coreg']:
+                    scale = DEFAULTS['coreg'][ch_type + '_scale'] * 1e3
+            scale *= 1e-3 if self._units == 'm' else 1
+            # plot sensors
+            if isinstance(ch_coord, tuple):  # is meg, plot coil
+                verts, triangles = ch_coord
+                self._renderer.surface(
+                    surface=dict(rr=verts * 1e3, tris=triangles),
+                    color=tuple(color[:3]), opacity=alpha,
+                    backface_culling=True)
+            else:
+                self._renderer.sphere(
+                    center=tuple(ch_coord * 1e3), color=color, scale=scale)
+            if ch_name in sources and fnirs_sources:
+                self._renderer.sphere(
+                    center=tuple(sources[ch_name] * 1e3),
+                    color=colorConverter.to_rgba(fnirs_sources, alpha),
+                    scale=scale)
+            if ch_name in detectors and fnirs_detectors:
+                self._renderer.sphere(
+                    center=tuple(detectors[ch_name] * 1e3),
+                    color=colorConverter.to_rgba(fnirs_detectors, alpha),
+                    scale=scale)
+            if ch_name in sources and ch_name in detectors and fnirs_pairs:
+                self._renderer.tube(  # array of origin and dest points
+                    origin=sources[ch_name][np.newaxis] * 1e3,
+                    destination=detectors[ch_name][np.newaxis] * 1e3,
+                    color=colorConverter.to_rgba(fnirs_pairs, alpha),
+                    radius=scale / 5)
+
+        if meg_helmet and pick_types(info.copy(), meg=True).size > 0:
+            surf = get_meg_helmet_surf(info, head_mri_t)
+            verts, triangles = surf['rr'] * 1e3, surf['tris']
+            color = _assign_by_name_or_ch_type(colors, 'meg_helmet', 'meg')
+            if color is None:
+                color = DEFAULTS['coreg']['helmet_color']
+            alpha = _assign_by_name_or_ch_type(alphas, 'meg_helmet', 'meg')
+            alpha = 0.25 if alpha is None else alpha
+            self._renderer.mesh(*verts.T, triangles,
+                                color=colorConverter.to_rgba(color, alpha),
+                                opacity=alpha, reset_camera=False,
+                                render=False)
+
+        self._renderer._update()
```

Comment:
this block of code above is not too redundant with the code in plot_alignment?

@GuillaumeFavelier what do you think?

Reply:
It's a bit redundant but the API is a bit different so I think that makes it difficult to entirely avoid the redundancy.

Reply:
I don't see why. If you have a private function that takes a renderer and the parameters it needs it should be possible to add sensor data to any rendered in the right coordinate system. This is a big block of complex code.

Reply:
I agree with @agramfort, to me these should share the same code structure. The only difference really is that here the sensors are always plotted in the MRI coordinate frame, so this would just use a subset of some private function like:
```
def _plot_sensors(renderers, info, trans, meg, eeg, fnirs, coord_frame='mri', units='m'):
```
function or so. The `units` in `plot_alignment` are always `'m'`, and the `coord_frame` in `Brain.add_sensors` will always be `'mri'`, but otherwise both can just call this function with the correct arguments (one renderer for `plot_alignment` and potentially multiple for `Brain`) and get identical visualizations (and DRY code) that way...

Reply:
+1